whitelistcli/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use clap::Parser;
+use std::fs::File;
+use std::str::FromStr;
+use ethers::prelude::{Address, Middleware, Provider};
+use ethers_providers::{Ws};
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt::config::Hasher;
+use subxt::config::substrate::BlakeTwo256;
+use subxt::ext::sp_core::bytes::from_hex;
+use subxt::ext::sp_core::{Pair, sr25519};
+use subxt::tx::PairSigner;
+use subxt::utils::{AccountId32, H160};
+
+mod api;
+
+#[derive(Parser)]
+struct Cli {
+	node_url: String,
+	private_key: String,
+	file_path: String,
+}
+
+#[derive(Debug, serde::Deserialize)]
+struct Record {
+	substrate_address: String,
+	eth_address: String,
+	amount: u128,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+	let args = Cli::parse();
+
+	let substrate_client = OnlineClient::<PolkadotConfig>::from_url(args.node_url.clone()).await?;
+	let eth_provider = Provider::<Ws>::new(Ws::connect(args.node_url).await?);
+
+	let decoded_sk = from_hex(&args.private_key)?;
+	let pair = sr25519::Pair::from_seed_slice(&decoded_sk)?;
+	let pair_signer = PairSigner::new(pair);
+
+	let file = File::open(args.file_path)?;
+	let mut rdr = csv::Reader::from_reader(file);
+
+	let mut eth_accs = Vec::new();
+
+	for result in rdr.deserialize() {
+		let record: Record = result?;
+		println!("record {:?}", record);
+
+		let substrate_account = AccountId32::from_str(&record.substrate_address)?;
+		let eth_account = record.eth_address.parse::<Address>()?;
+
+		eth_accs.push(eth_account);
+
+		process(
+			&substrate_client,
+			&pair_signer,
+			substrate_account,
+			eth_account,
+			record.amount,
+		).await?;
+	}
+
+	for acc in eth_accs {
+		let balance = eth_provider.get_balance(acc, None).await?;
```

Comment:
I don't think it's not helpful. Maybe report errors if it mismatch, but to output it as is not practical, I guess.

* My bet either to add proper logging with verification that the balance is correct
* Or remove it and get rid of ethers at all.